src/pages/index.js

```diff
@@ -6,74 +6,134 @@ import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
 import "../pages/index.css";
 import {
-  initialCards,
+  //initialCards,
   config,
   profileEditForm,
   addCardFormElement,
   profileEditButton,
   addNewCardButton,
   profileTitleInput,
   profileDescriptionInput,
+  profileAvatarButton,
+  editAvatarForm,
+  //cardListEl,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupConfirmDelete from "../components/popupConfirmDelete.js";
+//----------------------------------API--------------------------------------//
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "096eb059-8249-432c-aed5-ea6f0d18c1d5",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+let cardSection;
+api
+  .getInitialCards()
+  .then((cardData) => {
+    cardSection = new Section(
+      {
+        items: cardData,
+        renderer: (cardData) => {
+          const cardElement = getCardElement(cardData);
+          cardSection.addItem(cardElement);
+        },
+      },
+      ".gallery__cards"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+api
+  .getUser()
+  .then((inputValues) => {
+    userInfo.setUserInfo(inputValues.name, inputValues.about);
+    userInfo.setAvatar(inputValues.avatar);
+  })
+  .catch((err) => console.error(err));
+
+function handleAddCardFormSubmit(cardData) {
+  newCardPopup.viewLoading(true);
+  api
+    .addCard(cardData)
+    .then((data) => {
+      cardSection.addItem(getCardElement(data));
+      addCardValidator.disableButton();
+      newCardPopup.resetForm();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      newCardPopup.viewLoading(false);
+    });
+}
 
 //----------------------------event handlers---------------------------------//
 function handleImageClick(cardData) {
   popupImage.open(cardData);
 }
 
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.name, inputValues.job);
-  profileEditValidator.disableButton();
-  profileEditPopup.close();
-}
-
-function handleAddCardFormSubmit(cardData) {
-  cardSection.addItem(
-    getCardElement({ name: cardData.name, link: cardData.link })
-  );
-  addCardValidator.disableButton();
-  newCardPopup.resetForm();
-  newCardPopup.close();
+  profileEditPopup.viewLoading(true);
+  api
+    .editProfile(inputValues.name, inputValues.about)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      profileEditValidator.disableButton();
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      avatarEditPopup.viewLoading(false);
```

Comment:
You need to correctly return the button text of the profile form in `finally`


`profileEditPopup` should be used